### PR TITLE
BaSP-TG-33: Test Timesheets: put, delete

### PR DESCRIPTION
**Pull request by FranciscoSantar**
Create Tests units for Update and Delete methods of time sheets route

### CODE REVIEW

File: src/test/timesheets.test.js [hunk 1]
```diff
@@ -1 +1,110 @@
...
+  });
+  test('Test error with wrong content update (wrong id)', async () => {
+    const response = await request(app).put(`/time-sheets/${timeSheetId}`).send(wrongMockedTimeSheet);
+    expect(response.body.data).toBe(undefined);
+  });
+  test('Test error with wrong content update (wrong id)', async () => {
+    const response = await request(app).put(`/time-sheets/${timeSheetId}`).send(wrongMockedTimeSheet);
+    expect(response.body.message[0].message).toContain('"task" is required');
+    expect(response.body.message[1].message).toContain('maximum 12 hours');
+  });
```
Comment (MaicolFreire001) [inline src/test/timesheets.test.js:67]: We could do the four expects in the same test

File: src/test/timesheets.test.js [hunk 2]
```diff
@@ -1 +1,110 @@
...
+
+  test('Test status code failure update (no employee found)', async () => {
+    const response = await request(app).put(`/time-sheets/${fakeTimeSheetId}`).send(correctMockedTimeSheet);
+    expect(response.status).toBe(404);
+  });
+  test('Test error in failure update (wrong id)', async () => {
+    const response = await request(app).put(`/time-sheets/${fakeTimeSheetId}`).send(correctMockedTimeSheet);
+    expect(response.body.error).toBeTruthy();
+  });
+});
```
Comment (MaicolFreire001) [inline src/test/timesheets.test.js:77]: We could put the two expects in the same test and also the test names doesn't match

File: src/test/timesheets.test.js [hunk 3]
```diff
@@ -1 +1,110 @@
...
+    expect(response.status).toBe(400);
+  });
+  test('Test error in failure delete (wrong id)', async () => {
+    const response = await request(app).delete(`/time-sheets/${wrongTimeSheetId}`).send();
+    expect(response.body.error).toBeTruthy();
+  });
+  test('Test error messagge contain id in failure delete (wrong id)', async () => {
+    const response = await request(app).delete(`/time-sheets/${wrongTimeSheetId}`).send();
+    expect(response.body.message).toContain(`${wrongTimeSheetId}`);
+  });
```
Comment (MaicolFreire001) [inline src/test/timesheets.test.js:96]: We could do the three expects in the same test

File: src/test/timesheets.test.js [hunk 4]
```diff
@@ -1 +1,110 @@
...
+    expect(response.status).toBe(404);
+  });
+  test('Test error in failure delete (wrong id)', async () => {
+    const response = await request(app).delete(`/time-sheets/${fakeTimeSheetId}`).send();
+    expect(response.body.error).toBeTruthy();
+  });
+  test('Test error messagge contain id in failure delete (wrong id)', async () => {
+    const response = await request(app).delete(`/time-sheets/${fakeTimeSheetId}`).send();
+    expect(response.body.message).toContain(`${fakeTimeSheetId}`);
+  });
```
Comment (MaicolFreire001) [inline src/test/timesheets.test.js:109]: We could do the three expects in the same test and also the test names doesn't match

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

File: src/test/timesheets.test.js [hunk 5]
```diff
@@ -1 +1,80 @@
...
+  employee: mongoose.Types.ObjectId('63531244ec6456efd12685ef'),
+  hours: 15,
+};
+
+describe('UPDATE /timesheet', () => {
+  test('Test on a success update', async () => {
+    const response = await request(app).put(`/time-sheets/${timeSheetId}`).send(correctMockedTimeSheet);
+    expect(response.status).toBe(200);
+    expect(response.body.data).toBeDefined();
+    expect(response.body.data.hours).toBe(10);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:37]: We should also test the message and the error of the body in the response. And there is no need to test if the data has the correct hours

File: src/test/timesheets.test.js [hunk 6]
```diff
@@ -1 +1,80 @@
...
+  });
+
+  test('Test on a update with a incorrect format id', async () => {
+    const response = await request(app).put(`/time-sheets/${wrongTimeSheetId}`).send(correctMockedTimeSheet);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:43]: We should also test the message of the body

File: src/test/timesheets.test.js [hunk 7]
```diff
@@ -1 +1,80 @@
...
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBe(undefined);
+    expect(response.body.message[0].message).toContain('"task" is required');
+    expect(response.body.message[1].message).toContain('maximum 12 hours');
+  });
+
+  test('Test on a update with an non existing employee', async () => {
+    const response = await request(app).put(`/time-sheets/${fakeTimeSheetId}`).send(correctMockedTimeSheet);
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:57]: We should also test the message of the body

File: src/test/timesheets.test.js [hunk 8]
```diff
@@ -1 +1,80 @@
...
+  test('Test of a success delete', async () => {
+    const response = await request(app).delete(`/time-sheets/${timeSheetId}`).send();
+    expect(response.status).toBe(204);
+  });
+
+  test('Test of a failure delete (wrong id)', async () => {
+    const response = await request(app).delete(`/time-sheets/${wrongTimeSheetId}`).send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toContain(`${wrongTimeSheetId}`);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:71]: We should use the method toBe()

File: src/test/timesheets.test.js [hunk 9]
```diff
@@ -1 +1,80 @@
...
+  });
+
+  test('Test of a failure delete (wrong id)', async () => {
+    const response = await request(app).delete(`/time-sheets/${fakeTimeSheetId}`).send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toContain(`${fakeTimeSheetId}`);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:78]: We should use the method toBe()

File: src/test/timesheets.test.js [hunk 10]
```diff
@@ -1 +1,80 @@
...
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Test on a update with wrong time sheet', async () => {
+    const response = await request(app).put(`/time-sheets/${timeSheetId}`).send(wrongMockedTimeSheet);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBe(undefined);
+    expect(response.body.message[0].message).toContain('"task" is required');
+    expect(response.body.message[1].message).toContain('maximum 12 hours');
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:51]: We should use the method toBe()

File: src/test/timesheets.test.js [hunk 11]
```diff
@@ -1 +1,80 @@
...
+  test('Test on a update with a incorrect format id', async () => {
+    const response = await request(app).put(`/time-sheets/${wrongTimeSheetId}`).send(correctMockedTimeSheet);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Test on a update with wrong time sheet', async () => {
+    const response = await request(app).put(`/time-sheets/${timeSheetId}`).send(wrongMockedTimeSheet);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBe(undefined);
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:49]: We could use .not.toBeDefined(). So we use the same method as before to tests for defined and undefined.

File: src/test/timesheets.test.js [hunk 12]
```diff
@@ -1 +1,80 @@
...
+    expect(response.body.message[0].message).toContain('"task" is required');
+    expect(response.body.message[1].message).toContain('maximum 12 hours');
+  });
```
Comment (nicolasblasco) [inline src/test/timesheets.test.js:52]: We should also test the error of the body to be true

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

**Review by MaicolFreire001 — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**